scipy/stats/ztest.py

```diff
@@ -0,0 +1,225 @@
+# shall Gary Strangman be liable for any direct, indirect, incidental,
+# special, exemplary or consequential damages (including, but not limited
+# to, loss of use, data or profits, or business interruption) however
+# caused and on any theory of liability, whether in contract, strict
+# liability or tort (including negligence or otherwise) arising in any way
+# out of the use of this software, even if advised of the possibility of
+# such damage.
+
+"""
+A collection of basic statistical tests for Python to run a Z-test.
+
+References
+----------
+.. [CRCProbStat2000] Zwillinger, D. and Kokoska, S. (2000). CRC Standard
+   Probability and Statistics Tables and Formulae. Chapman & Hall: New
+   York. 2000.
+
+"""
+import warnings
+import math
+from math import gcd
+from collections import namedtuple
+from itertools import permutations
+
+import numpy as np
+from numpy import array, asarray, ma
+
+from scipy.spatial.distance import cdist
+from scipy.ndimage import measurements
+from scipy._lib._util import (check_random_state, MapWrapper,
+                              rng_integers, float_factorial)
+import scipy.special as special
+from scipy import linalg
+from . import distributions
+from . import mstats_basic
+from ._stats_mstats_common import (_find_repeats, linregress, theilslopes,
+                                   siegelslopes)
+from ._stats import (_kendall_dis, _toint64, _weightedrankedtau,
+                     _local_correlations)
+from dataclasses import make_dataclass
+
+
+# Functions/classes in other files should be added in `__init__.py`, not here
+__all__ = ['find_repeats', 'gmean', 'hmean', 'mode', 'tmean', 'tvar',
+           'tmin', 'tmax', 'tstd', 'tsem', 'moment', 'variation',
+           'skew', 'kurtosis', 'describe', 'skewtest', 'kurtosistest',
+           'normaltest', 'jarque_bera', 'itemfreq',
+           'scoreatpercentile', 'percentileofscore',
+           'cumfreq', 'relfreq', 'obrientransform',
+           'sem', 'zmap', 'zscore', 'iqr', 'gstd', 'median_absolute_deviation',
+           'median_abs_deviation',
+           'sigmaclip', 'trimboth', 'trim1', 'trim_mean',
+           'f_oneway', 'F_onewayConstantInputWarning',
+           'F_onewayBadInputSizesWarning',
+           'PearsonRConstantInputWarning', 'PearsonRNearConstantInputWarning',
+           'pearsonr', 'fisher_exact',
+           'SpearmanRConstantInputWarning', 'spearmanr', 'pointbiserialr',
+           'kendalltau', 'weightedtau', 'multiscale_graphcorr',
+           'linregress', 'siegelslopes', 'theilslopes', 'ttest_1samp',
+           'ttest_ind', 'ttest_ind_from_stats', 'ttest_rel',
+           'kstest', 'ks_1samp', 'ks_2samp',
+           'chisquare', 'power_divergence', 'mannwhitneyu',
+           'tiecorrect', 'ranksums', 'kruskal', 'friedmanchisquare',
+           'rankdata',
+           'combine_pvalues', 'wasserstein_distance', 'energy_distance',
+           'brunnermunzel', 'alexandergovern']
+_one_sample_z_table = {
```

Comment:
Can you add a comment about where these values come from?

Also, is 4 digits precise enough, and is there not a way to calculate these values instead?

Reply:
These are the cdf of a normal.  Why not use `stats.norm.cdf`?